OctopusDSC/DSCResources/cOctopusServer/cOctopusServer.psm1

```diff
@@ -807,6 +843,42 @@ function Set-OctopusDeployConfiguration {
             throw "'OctopusBuiltInWorkerCredential' is only supported from Octopus 2018.1.0 and newer."
         }
     }
+
+    if (($SSLPort -ne 443) -and (!$SSLWebListenPrefix.Contains(":$SSLPort")))
+    {
+        throw "Please update the value for SSLWebListenPrefix to use port $SSLPort."
+    }
+
+    if (![string]::IsNullOrEmpty($SSLCertificateThumbprint) -and ![string]::IsNullOrEmpty($SSLCertificateStoreName))
```

Comment:
This should be checking if the values have changed, otherwise it will be slow as it's resetting values that haven't changed.